nifi-commons/nifi-utils/src/main/java/org/apache/nifi/processor/util/FlowFileFilters.java

```diff
@@ -43,23 +43,20 @@ public static FlowFileFilter newSizeBasedFilter(final double maxSize, final Data
 
             @Override
             public FlowFileFilterResult filter(final FlowFile flowFile) {
-                if (count == 0) {
-                    count++;
-                    size += flowFile.getSize();
+                count += 1;
+                size += flowFile.getSize();
 
+                if (count == 1) {
+                    // first FlowFile is always accepted
                     return FlowFileFilterResult.ACCEPT_AND_CONTINUE;
                 }
 
-                if ((size + flowFile.getSize() > maxBytes) || (count + 1 > maxCount)) {
+                if (size > maxBytes || count > maxCount) {
```

Comment:
In the previous code count and size were not changed for this case.  The case of reject and terminate is terminal and this FlowFileFilterResult will not be further used nor those values in anyway referenced?

Reply:
I'm not sure I correctly understood your comment. Please clarify if the following doesn't address your point.

My goal was to simplify the implementation by unifying the three points of computation (inside the first `if` statement, inside the predicate of the second `if`, and before the last return) into a single point.
I added test cases for a few selected scenarios before adjusting the implementation to ensure that the behavior remains the same.

The JavaDoc states:
>  Returns a new {@link FlowFileFilter} that will pull FlowFiles until the maximum file size has been reached, or the maximum FlowFile Count was been reached ...

Therefore, once either the maximum count or accumulated size limit is reached, all subsequent FlowFiles should be rejected.

Upon closer inspection, the old implementation contained a minor bug related to this behavior. If a FlowFile exceeding the size limit was initially rejected (with `REJECT_AND_TERMINATE`), and the caller subsequently passed an empty FlowFile to the filter, the filter would incorrectly return `ACCEPT_AND_CONTINUE` even though the limit had already been reached. While this behavior is technically incorrect, it's primarily a consequence of the caller not adhering to the `FlowFileFilterResult` contract and thus not a critical bug.
I enhanced one of the test cases to demonstrate this behavior.

Both the old and new implementations are susceptible to potential integer overflow issues for both `count` and `size`. I'm not sure if that's something worth addressing though.

I'm content with leaving the implementation of `newSizeBasedFilter` unchanged if that's prefered. I encountered the code while working on the desired changes for `PackageFlowFile` and deemed it worth some tests / minor refactor.